ci-operator/step-registry/telcov10n/jiras/telcov10n-jiras-commands.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ telcov10n Fix user IDs in a container ************"
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function gathering_jiras_info {
+
+  echo "************ telcov10n Gathering Jiras info ************"
+
+  echo
+  echo "Jiras to watch out"
+  echo
+
+  jq -c '.[]' <<< "$(yq -o json <<< ${JIRAS})" | while read -r entry; do
+    # Extract the filename and content
+    description=$(echo "$entry" | jq -r '.description')
+    links=$(echo "$entry" | jq -r '.links')
+    pull_requets=$(echo "$entry" | jq -r '.PRs')
+
+     process_jiras "${description}" "${links}" "${pull_requets}"
+  done
+}
+
+function process_jiras {
+
+  desc=$1 ; shift
+  jiras=$1 ; shift
+  prs=$1
+
+  closed_jiras "${desc}" "${jiras}" "${prs}"
+
+  echo "####################################################################################################"
+  echo " -- Description ------------------------------------------------------------------------------------"
+  echo
+  echo "${desc}"
+  echo
+  echo " -- JIRAS ------------------------------------------------------------------------------------------"
+  echo
+  echo "${jiras}" | jq -r '.[]'
+  echo
+  echo " -- Related Pull Requests --------------------------------------------------------------------------"
+  echo
+  echo "${prs}" | jq -r '.[]'
+  echo
+  echo "####################################################################################################"
+  echo
+
+}
+
+function closed_jiras {
+
+  # TODO: Send a Slack notification or email to signal that a PreGA workaround
+  # should be reviewed
+  echo
```

Comment:
@natifridman ,

I have tried to implement something very naive but good enough to track PreGA missing components. If any of you have an idea to extend it so that a notification can be sent in the even of a tracked Jira is closed, that would be of great help. WDYT?

Reply:
I'm not sure I fully understand what you are looking for, but if you need to send Slack notifications you can use [webhooks](https://api.slack.com/messaging/webhooks)

Reply:
Actually two things are needed. The slack webhooks and Jira API Rest access via Token to check out whether the related Jira has been closed and it is time to revert the workaround so that the job will continue testing as much final GA artefacts as possible.

For now, this is just one way to follow up on what you commented. At least this way we will know what we have been changing so that the Job can work with the PreGA builds.

Reply:
Maybe it's something we can do manually by following the Jira issue to get a notification? not sure if it's something that needs automation here...

Reply:
Ok. Let's agree on the following procedure then:
- Find out the related Jira (or create a new one to track if no one exist) and add yourself to the watchers group.
-  Create an new entry in the Job config like the one...

```
JIRAS: |-
  - description: OCP 4.19 failing to deploy due to bootkube.sh error, missing oc command
    links:
      - https://issues.redhat.com/browse/OCPBUGS-50585
    PRs:
      - https://github.com/openshift/release/pull/61568
```

  ... where you put the tracking info needed to revert the WA along with the WA PR itself.
  - Finally, at info will be dumped in the Job's logs.
  
  Make sense?
  
  I am going to rework this commit to remove the automation proposal.

Reply:
Sounds good